src/components/SurveyComponent.vue

```diff
@@ -1,324 +1,277 @@
+      // Optionally request Observations using categories
+      if (
+        process.env.VUE_APP_FHIR_OBSERVATION_CATEGORY_QUERIES &&
+        process.env.VUE_APP_FHIR_OBSERVATION_CATEGORY_QUERIES.toLowerCase() == 'true') {
+        getObservationCategories().forEach(cat => {
+          observationQueryString = observationQueryString + '&category=' + cat;
+        });
       }
-    });
+      const requests = [
+        client.request('/Condition?patient=' +  this.patientId),
+        client.request(observationQueryString),
+        client.request('/Procedure?patient=' +  this.patientId),
+        client.request('/QuestionnaireResponse?patient=' +  this.patientId)
+      ];
+      //get all resources
+      Promise.all(requests).then(results => {
+        results.forEach(result => {
+          if (!result) return true;
+          if (result.resourceType == 'Bundle' && result.entry) {
+            result.entry.forEach(o => {
+              if (o && o.resource) this.patientBundle.entry.push({resource: o.resource});
+            });
+          } else if (Array.isArray(result)) {
+            result.forEach(o => {
+              if (o.resourceType) this.patientBundle.entry.push({resource: o});
+            });
+          } else {
+            this.patientBundle.entry.push({resource: result});
+          }
+        });
+      });
+    }, //
+    getQuestionnaireURL() {
+      if (!this.questionnaire) return null;
+      //used to pair questionaire with responses
+      return this.questionnaire.url;
+    },
+    setQuestionnaireAuthor() {
+      // Add the `subject` element to the QuestionnaireResponse
+      this.questionnaireResponse.subject = {
+        reference: `Patient/${this.patientId}`
+      };
 
-    // Get any Procedure resources
-    await client.request('/Procedure?patient=' + pid).then(function(pr) {
-      if (pr) {
-        if (pr.resourceType == 'Bundle' && pr.entry) {
-          pr.entry.forEach(p => {
-            if (p.resource) patientBundle.entry.push({resource: p.resource});
-          });
-        } else if (Array.isArray(pr)) {
-          pr.forEach(p => {
-            if (p.resourceType) patientBundle.entry.push({resource: p});
-          });
-        } else {
-          patientBundle.entry.push({resource: pr});
+      // Record who is entering and submitting the responses
+      // How do we know the author here?
+      let questionnaireAuthor = process.env.VUE_APP_QUESTIONNAIRE_AUTHOR && process.env.VUE_APP_QUESTIONNAIRE_AUTHOR.toLowerCase();
+      if (questionnaireAuthor == 'practitioner') {
+        // Only add the `author` element if we can get the user id from the client
+        if (client.user && client.user.fhirUser) {
+          this.questionnaireResponse.author = {
+            reference: client.user.fhirUser
+          };
+        }
+      } else if (questionnaireAuthor == 'patient') {
+        this.questionnaireResponse.author = {
+          reference: this.questionnaireResponse.subject.reference
```

Comment:
Aside: I'm really surprised they have a config to control this. SoF should allow you to detect this information. Maybe they were in a non-compliant EHR?